### PR TITLE
First draft of the report feature

### PR DESCRIPTION
**Pull request by SwiftPredator**

### CODE REVIEW

File: catalogue/migrations/0006_report.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Generated by Django 3.2.4 on 2021-08-13 12:09
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('catalogue', '0005_feedback'),
```
Comment (fjen) [inline catalogue/migrations/0006_report.py:10]: should come after current 0006 migration

File: catalogue/views.py [hunk 2]
```diff
@@ -121,3 +121,46 @@ class ComparisonView(FilterView):
...
+
+
+class ReportView(generic.detail.SingleObjectMixin, generic.edit.FormView):
+    template_name = "catalogue/modals/report/form.html"
+    success_template = "catalogue/modals/report/success.html"
+    form_class = ReportForm
+    model = Component
+
+    def get(self, request, *args, **kwargs):
+        self.object = self.get_object()
```
Comment (fjen) [inline catalogue/views.py:133]: is this necessary?

File: catalogue/views.py [hunk 3]
```diff
@@ -121,3 +121,46 @@ class ComparisonView(FilterView):
...
+        return super().get(request, *args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        self.request = request
```
Comment (fjen) [inline catalogue/views.py:137]: self.request should be filled with request anyway

File: catalogue/views.py [hunk 4]
```diff
@@ -121,3 +121,46 @@ class ComparisonView(FilterView):
...
+        self.object = self.get_object()
+        return super().post(request, *args, **kwargs)
+
+    def form_valid(self, form):
+        report = form.save(commit=False)
+        report.component = self.object
```
Comment (fjen) [inline catalogue/views.py:143]: `self.get_object` doesn't work here?

**Review by fjen — CHANGES_REQUESTED** [review]

File: catalogue/views.py (hunk 3 above)
Comment (fjen) [reply catalogue/views.py:137]: see https://github.com/django/django/blob/main/django/views/generic/base.py#L89

File: catalogue/views.py (hunk 2 above)
Comment (SwiftPredator) [reply catalogue/views.py:133]: Seems to be necessary. When I remove it, it doesn't work anymore. Holds for post and get. The super class seems to need this attribute. Also tried to remove the post and get methods completely.
Message: 
  File "/Users/ironmonkeyapps/Desktop/Studium/Master/Semester2/solutionCatalogue/catalogue/views.py", line 156, in get_context_data
    context = super().get_context_data(**kwargs)
  File "/Users/ironmonkeyapps/.local/share/virtualenvs/solutionCatalogue--U9CtuRt/lib/python3.9/site-packages/django/views/generic/detail.py", line 94, in get_context_data
    if self.object:
AttributeError: 'ReportView' object has no attribute 'object'

Comment (fjen) [reply catalogue/views.py:124]: alright. 

> The base implementation of this method requires that the self.object attribute be set by the view (even if None). Be sure to do this if you are using this mixin without one of the built-in views that does so.

https://docs.djangoproject.com/en/3.2/ref/class-based-views/mixins-single-object/#django.views.generic.detail.SingleObjectMixin.get_context_data

File: catalogue/admin.py [hunk 5]
```diff
@@ -293,6 +293,48 @@ def message_short(self, obj):
     def get_search_url(self, obj):
         return mark_safe(f"""<a href='{obj.search_url}'>{obj.search_url}</a>""")
 
+    get_search_url.short_description = Feedback._meta.get_field(
+        "search_url"
+    ).verbose_name
+
+    message_short.short_description = Feedback._meta.get_field("message").verbose_name
```
Comment (fjen) [inline catalogue/admin.py:300]: This was removed in the main branch, please rebase

**Comment by fjen** [comment]
merged manually

**Pull request closed without merging**